src/Storages/ObjectStorageQueue/ObjectStorageQueueSettings.cpp

```diff
@@ -21,16 +21,28 @@ void ObjectStorageQueueSettings::loadFromQuery(ASTStorage & storage_def)
     {
         try
         {
+            std::vector<std::string> ignore_settings;
+            auto settings_changes = storage_def.settings->changes;
```

Comment:
I think now it was a mistake that previously it modified actual settings, because it was affecting the AST of the query which is used for metadata file, which might have unexpected consequences with table definition mismatch.